src/terms/hartree.jl

```diff
@@ -1,3 +1,5 @@
+using GPUArraysCore
```

Comment:
I would move this to the general DFTK.jl

My rule of thumb for the `using`s is:
- Is it a package that I only use in this (or a few selected) files or where the functionality of the package is closely coupled to one file in DFTK, then I put the `using` into that file.
- If it's more "global" and affects everyone (like the `allowscalar`) that the package should be in the `DFTK.jl` file.

We don't follow this rule thoroughly, though.